### PR TITLE
New guides for AutoML and Shap explainability

### PR DESCRIPTION
**Pull request by KarthikDutt**
Adding first draft of the guides for AutoML (For tabular data) and Model explainability using SHAP for tabular data
-----

# Checklist

Please go through each entry in the below checklist and mark an 'X' if that condition has been met. Every entry should be marked with an 'X' to be get the Pull Request approved.

- [ ] All `import`s are in the first cell? First block of imports are standard libraries, second block are 3rd party libraries, third block are all `arcgis` imports? Note that in some cases, for samples, it is a good idea to keep the imports next to where they are used, particularly for uncommonly used features that we want to highlight.
- [ ] All `GIS` object instantiations are one of the following?
    - `gis = GIS()`
    - `gis = GIS('https://www.arcgis.com', 'arcgis_python', 'P@ssword123')`
    - `gis = GIS(profile="your_online_profile")`
    - `gis = GIS('https://pythonapi.playground.esri.com/portal', 'arcgis_python', 'amazing_arcgis_123')`
    - `gis = GIS(profile="your_enterprise_portal")`
- [ ] If this notebook requires setup or teardown, did you add the appropriate code to `./misc/setup.py` and/or `./misc/teardown.py`?
- [ ] If this notebook references any portal items that need to be staged on AGOL/Python API playground, did you coordinate with a Python API team member to stage the item the correct way with the api\_data\_owner user?
- [ ] Code refactored & split out across multiple cells, useful comments?
- [ ] Consistent voice/tense/narrative style? Thoroughly checked for typos?
- [ ] All images used like `` instead of ``? All map widgets contain a static image preview? (Call `mapview_inst.take_screenshot()` to do so)
- [ ] All file paths are constructed in an OS-agnostic fashion with `os.path.join()`? (Instead of `r"\foo\bar"`, `os.path.join(os.path.sep, "foo", "bar")`, etc.)
- [ ] **IF YOU WANT THIS SAMPLE TO BE DISPLAYED ON THE DEVELOPERS.ARCGIS.COM WEBSITE**, ping @ DavidJVitale so he can add it to the list for the next deploy

### CODE REVIEW

File: samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,534 @@
+{
```
Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Keep in mind the growing uset need --&gt; typo in word 'user'</p><p>Change sentence to "To support the growing need to make models more explainable, <code>arcgis.learn</code>  ..."</p><p>fully conneccted -&gt; typo in connected</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    import shap
</pre><p>Should we provide steps to install shap?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>we might want to change the login to <code>gis=GIS('home')</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    calgary_no_southland_solar
</pre><p>This line is not needed</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    calgary_no_southland_solar_layer_sdf = calgary_no_southland_solar_layer.query().sdf
</pre><p>better to split the cell before this line, so the map preview will show.</p><p>Can you take a screenshot of the map and add it?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p> method in <code>arcgis.learn</code> <strong>for </strong>MLModel and Fully.. (missing for)</p><p>The paramter explain -&gt; typo in parameter. Better to quote 'explain' in this line as well. same in next line / bullet.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>typo in explai_index</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>the mean average value  -&gt; redundant, either say mean value or average value.</p><p>Apart from <strong>that, </strong>this plot ....</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Similar to feature importance in sklearn) - &gt; no need to cap S in Similar. Can you add a hyperlink to the said topic in sklearn?</p><p>This plot tells the -&gt; The plot below (for clarity)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

File: samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb [hunk 2]
```diff
@@ -0,0 +1,1201 @@
+{
```
Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>Model selection -&gt; lower case M</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>feature layer hosted on the <em>geosaurus</em> -&gt; ArcGIS Online</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>Better to change login to <code>gis=GIS('home')</code>.  Then ensure the data layer is shared publicly</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>the HTML reports are not embedded. Can you add screenshots instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>This path with break on user's end and also on AGS Notebooks. Can you replace the it something that is relative to this notebook?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (AtmaMani) [inline samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>same here, the HTML reports are not embedded. Can you add screenshots instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/977/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

**Review by AtmaMani — CHANGES_REQUESTED** [review]
@KarthikDutt Nice set of notebooks. I have requested changes via reviewnb app.
I think the model explainability notebook could be made into a guide instead of a sample notebook

File: samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb (hunk 1 above)
Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>SHAP will come with the conda metapackage and explicit installation would not be necessary.</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>SHAP will come with the conda metapackage and explicit installation would not be necessary.</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p><p> </p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p><p> </p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb:1]: <p>Done</p>

File: samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb (hunk 2 above)
Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>Done</p>

Comment (KarthikDutt) [reply samples/04_gis_analysts_data_scientists/tabular_data_supervised_learning_using_automl.ipynb:1]: <p>Removed this line. It is not needed again. </p>

**Comment by KarthikDutt** [comment]
@AtmaMani - Thanks for the review comments. I have incorporated them.

File: samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb (hunk 1 above)
Comment (AtmaMani) [reply samples/04_gis_analysts_data_scientists/model_explainability_using_shap_for_tabular_data.ipynb]: <p>sounds good</p>

**Review by AtmaMani — APPROVED** [review]

**Pull request merged**